utils/build_swift/build_swift/driver_arguments.py

```diff
@@ -193,6 +193,11 @@ def _apply_default_arguments(args):
         args.test_swiftevolve = False
         args.test_toolchainbenchmarks = False
 
+    # --test implies --test-early-swift-driver
+    # (unless explicitly skipped with `--skip-test-early-swift-driver`)
+    if args.test and args.test_early_swift_driver is None:
```

Comment:
What happens here if we don't build swift driver? Please add tests for the failure modes here.

Reply:
Changed to only trigger `test_early_swift_driver` if early driver is being built.
If we expect early driver to get built, but the host toolchain did not have a Swift compiler, this will still run the driver test subset, it'll just run against the legacy driver.

Reply:
Seems like something we would want to warn/error to provide visibility. Consider this running on a bot and we just silently stopped testing early swift driver and were testing the legacy driver the whole time. That would be unfortunate. Can you add a warning in such a scenario at minimum?